### PR TITLE
fx.replace_pattern accepts pattern/replacement as GraphModule

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88479

Symbolic tracer is no longer the default tracer to produce fx graph. 
SubgraphRewriter should thus accept a raw GraphModule, rather than use symbolic tracer by default.

### CODE REVIEW

File: torch/fx/subgraph_rewriter.py [hunk 1]
```diff
@@ -217,8 +221,16 @@ def _replace_pattern(
 
     # Get the graphs for `gm`, `pattern`, `replacement`
     original_graph: Graph = gm.graph
-    pattern_graph: Graph = symbolic_trace(pattern).graph
-    replacement_graph: Graph = symbolic_trace(replacement).graph
+
+    if isinstance(pattern, GraphModule):
+        pattern_graph = pattern.graph
+    else:
+        pattern_graph = symbolic_trace(pattern).graph
```
Comment (jerryzh168) [inline torch/fx/subgraph_rewriter.py:228]: if we do symbolic_trace then the pattern can be a `nn.Module` I think?

**Review by jerryzh168 — APPROVED** [review]

File: torch/fx/subgraph_rewriter.py (hunk 1 above)
Comment (SherlockNoMad) [reply torch/fx/subgraph_rewriter.py:228]: lol... nn.Module is also a Callable, coz it implemented __call__ 
So we are good

Comment (jerryzh168) [reply torch/fx/subgraph_rewriter.py:228]: `GraphModule` is a subclass of `nn.Module` though

Comment (jerryzh168) [reply torch/fx/subgraph_rewriter.py:228]: sorry I think I misread the PR, sounds good

**Pull request closed without merging**